src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -956,7 +957,12 @@ internal static RefKind GetArgumentRefKind(ImmutableArray<BoundExpression> argum
                     argRefKind = argRefKindsOpt[i];
 
                     Debug.Assert(argRefKind == parameters[i].RefKind ||
-                            argRefKind == RefKindExtensions.StrictIn && parameters[i].RefKind == RefKind.In,
+                            parameters[i].RefKind switch
+                            {
+                                RefKind.In => argRefKind == RefKindExtensions.StrictIn,
+                                RefKind.RefReadOnlyParameter => argRefKind is RefKind.None or RefKind.In or RefKind.Ref or RefKindExtensions.StrictIn,
```

Comment:
>RefKind.None [](http://example.com/codeflow?start=78&length=12)

Judging based on ```EmitArgument``` code, it looks like ```RefKind.None``` won't be handled there properly for a parameter with ```RefKind.RefReadOnlyParameter```. Perhaps it should be removed from the list.

Reply:
Right. `Ref` can be also removed since `ref readonly` is always lowered to either `in` or "strict `in`", so I think we should assert that here.